packages/SwingSet/src/types-internal.js

```diff
@@ -34,16 +35,17 @@ export {};
  *   enableDisavow: boolean,
  *   useTranscript: boolean,
  *   name: string,
- *   compareSyscalls?: (originalSyscall: {}, newSyscall: {}) => Error | undefined,
+ *   compareSyscalls?: import('./kernel/vat-loader/transcript.js').CompareSyscalls,
  *   sourcedConsole: Pick<Console, 'debug' | 'log' | 'info' | 'warn' | 'error'>,
  *   enableSetup: boolean,
  *   setup?: unknown,
  *   bundle?: Bundle,
  * }} ManagerOptions
  *
  * @typedef { { deliver: (delivery: VatDeliveryObject) => Promise<VatDeliveryResult>,
+ *              replayOneDelivery: (delivery: VatDeliveryObject, expectedSyscalls: VatSyscallObject[], deliveryNum: number) => Promise<VatDeliveryResult>,
  *              replayTranscript: (startPos: number | undefined) => Promise<number?>,
- *              makeSnapshot?: (endPos: number, ss: SnapStore) => Promise<SnapshotResult>,
+ *              makeSnapshot?: undefined | ((endPos: number, ss: SnapStore) => Promise<SnapshotResult>),
```

Comment:
What does it mean to have both the `?` in `makeSnapsot?` *and* the `undefined |` ?

The original intention was that `manager.makeSnapshot` may or may not be present, but if it is, then it must be a function with the given signature.

Reply:
I don't remember why I had to do this, but I think tsc was complaining without it back in the original pismo work

I do remember something about `makeSnapshot` being provided as argument and unconditionally installed in the returned object even if `undefined`, so this is technically more accurate.